extensions/amp-a4a/0.1/test/test-refresh.js

```diff
@@ -0,0 +1,245 @@
+    });
+
+    it('should get refreshInterval from slot', () => {
+      expect(getPublisherSpecifiedRefreshInterval(
+          mockA4a.element, window, 'doubleclick')).to.equal(35000);
+    });
+
+    it('should get refreshInterval from meta tag', () => {
+      mockA4a.element.removeAttribute(DATA_ATTR_NAME);
+      const meta = window.document.createElement('meta');
+      meta.setAttribute('name', METATAG_NAME);
+      meta.setAttribute('content', 'doubleclick=40');
+      window.document.head.appendChild(meta);
+      expect(getPublisherSpecifiedRefreshInterval(
+          mockA4a.element, window, 'doubleclick')).to.equal(40000);
+    });
+
+    it('should call convertConfiguration_ and set proper units', () => {
+      const getConfigurationSpy = sandbox.spy(
+          RefreshManager.prototype, 'convertAndSanitizeConfiguration_');
+      const refreshManager = new RefreshManager(mockA4a, {
+        visiblePercentageMin: 50,
+        continuousTimeMin: 1,
+      }, 30000);
+      expect(getConfigurationSpy).to.be.calledOnce;
+      expect(refreshManager.config_).to.not.be.null;
+      expect(refreshManager.config_.visiblePercentageMin).to.equal(0.5);
+      expect(refreshManager.config_.continuousTimeMin).to.equal(1000);
+    });
+
+    describe('#ioCallback_', () => {
+      let refreshManager;
+      beforeEach(
+          () => refreshManager = new RefreshManager(mockA4a, config, 30000));
+
+      it('should stay in INITIAL state', () => {
+        const ioEntry = {
+          target: {
+            getAttribute: name => name == DATA_MANAGER_ID_NAME ? '0' : null,
+          },
+          intersectionRatio: refreshManager.config_.visiblePercentageMin,
+        };
+        refreshManager.ioCallback_([ioEntry]);
+        expect(refreshManager.visibilityTimeoutId_).to.be.null;
+        expect(refreshManager.state_).to.equal('initial');
+      });
+
+      it('should transition into VIEW_PENDING state', () => {
+        const ioEntry = {
+          target: mockA4a.element,
+          intersectionRatio: refreshManager.config_.visiblePercentageMin,
+        };
+        refreshManager.ioCallback_([ioEntry]);
+        expect(refreshManager.visibilityTimeoutId_).to.be.ok;
+        expect(refreshManager.state_).to.equal('view_pending');
+      });
+
+      it('should transition to VIEW_PENDING state then back to INITIAL', () => {
+        const ioEntry = {
+          target: mockA4a.element,
+          intersectionRatio: refreshManager.config_.visiblePercentageMin,
+        };
+        refreshManager.ioCallback_([ioEntry]);
+        expect(refreshManager.visibilityTimeoutId_).to.be.ok;
+        expect(refreshManager.state_).to.equal('view_pending');
+        ioEntry.intersectionRatio = 0;
+        refreshManager.ioCallback_([ioEntry]);
+        expect(refreshManager.visibilityTimeoutId_).to.be.null;
+        expect(refreshManager.state_).to.equal('initial');
+      });
+    });
+
+    it('should execute the refresh event correctly', () => {
```

Comment:
Looks like nothing above this line is changed in this file - please correct me if I am wrong.

Reply:
That's correct.

Reply:
@glevitzky This test is failing on master. Can you investigate and skip / fix the test?

https://travis-ci.org/ampproject/amphtml/jobs/373920148

Reply:
Sorry about that. Fix is here: https://github.com/ampproject/amphtml/pull/15019/.